src/index.js

```diff
@@ -1,36 +1,150 @@
+        const editInput = parent.querySelector('.edit--input');
+        const descPara = parent.querySelector('.desc');
+        const editForm = parent.querySelector('.edit-form');
+        parent.classList.add('hide--desc');
+        editInput.value = descPara.textContent;
+        editInput.focus();
+        editForm.addEventListener('submit', (e) => {
+          e.preventDefault();
+          if (editInput.value) {
+            this.editTask(Option, editInput.value);
+            this.populateToDoList();
+            this.setLocalStorage(this.myToDoList);
+          }
+        });
+      });
+    });
+    trashIcons.forEach((icon) => {
+      icon.addEventListener('click', (e) => {
+        const { Trash } = e.target.dataset;
+        this.removeTask(Trash);
+        this.updateTaskIndex();
+        this.populateToDoList();
+        this.setLocalStorage(this.myToDoList);
+      });
+    });
+    checkBoxes.forEach((checkBox) => {
+      checkBox.addEventListener('change', (e) => {
+        const { id } = e.target;
+        const parent = e.target.parentElement.parentElement;
+        const pTag = parent.querySelector('.desc');
+        pTag.classList.toggle('strike--through');
+        this.updateCompletionStatus(+id);
+        this.setLocalStorage(this.myToDoList);
+      });
+    });
+  };
+
+  addTask = (description, completed = false) => {
+    const index = this.myToDoList.length + 1;
+    const newTask = { description, completed, index };
+    this.myToDoList = [...this.myToDoList, newTask];
+  };
+
+  removeTask = (index) => {
+    this.myToDoList = this.myToDoList.filter((list) => list.index !== +index);
+  };
+
+  updateTaskIndex = () => {
+    this.myToDoList = this.myToDoList.map((list, index) => {
+      list.index = index + 1;
+      return list;
+    });
+  }
+
+  editTask = (index, value) => {
+    this.myToDoList = this.myToDoList.map((list) => {
+      if (list.index === +index) {
+        list.description = value;
+      }
+      return list;
+    });
+  };
+
+  updateCompletionStatus = (index) => {
+    this.myToDoList = this.myToDoList.map((list) => {
+      if (list.index === index) {
+        list.completed = !list.completed;
+      }
+      return list;
+    });
+  };
+
+  clearAllCompletedTasks = () => {
+    this.myToDoList = this.myToDoList.filter((list) => !list.completed);
+  };
+
+  addTaskToUI = () => {
+    form.addEventListener('submit', (e) => {
+      const inputList = document.querySelector('.input-list');
+      e.preventDefault();
+      const { value } = inputList;
+      if (value) {
+        this.addTask(value);
+        this.populateToDoList();
+        this.setLocalStorage(this.myToDoList);
+        inputList.value = '';
+      }
+    });
+  };
+
+  clearCompletedTasksFromUI = () => {
+    clearBtn.addEventListener('click', () => {
+      this.clearAllCompletedTasks();
+      this.setLocalStorage();
+      this.populateToDoList();
+    });
+  }
```

Comment:
- [ ] Kindly ensure that once the error causing tasks to not be added is resolved, the array of items stored in local storage should be updated after each user action, so the user gets the correct list values after the page reloads.

Reply:
Thank you so much for your review @Munsa1 , I have resolved the local storage error.